tests/featureWriters/markFeatureWriter_test.py

```diff
@@ -670,6 +676,71 @@ def test_mark_mkmk_features_with_GDEF(self, testufo):
             """
         )
 
+    def test_multiple_anchor_classes_base(self, FontClass):
+        dirname = os.path.dirname(os.path.dirname(__file__))
+        fontPath = os.path.join(dirname, "data", "MultipleAnchorClasses.ufo")
+        testufo = FontClass(fontPath)
+        generated = self.writeFeatures(testufo)
+
+        assert str(generated) == dedent(
+            """\
+            markClass acutecomb <anchor -175 589> @MC_topA;
+            markClass acutecomb <anchor -175 572> @MC_topE;
+
+            feature mark {
+                lookup mark2base {
+                    pos base a <anchor 515 581> mark @MC_topA;
+                } mark2base;
+
+                lookup mark2base_1 {
+                    pos base e <anchor -21 396> mark @MC_topE;
+                } mark2base_1;
+
+            } mark;
+            """
+        )
+
+    def test_multiple_anchor_classes_liga(self, FontClass):
+        ufo = FontClass()
+        liga = ufo.newGlyph("f_i")
+        liga.appendAnchor({"name": "top_1", "x": 100, "y": 500})
+        liga.appendAnchor({"name": "top_2", "x": 600, "y": 500})
+        ligaOther = ufo.newGlyph("f_f")
+        ligaOther.appendAnchor({"name": "topOther_1", "x": 101, "y": 501})
+        ligaOther.appendAnchor({"name": "topOther_2", "x": 601, "y": 501})
+        ligaMix = ufo.newGlyph("f_l")
+        ligaMix.appendAnchor({"name": "top_1", "x": 102, "y": 502})
+        ligaMix.appendAnchor({"name": "topOther_2", "x": 602, "y": 502})
```

Comment:
I'm particularly interested in knowing whether it's allowed to mix different anchor names in the same ligature like I did in that part of the test, and whether the proposed solution below is correct